modules/salt/files/bin/rename-wiki.py

```diff
@@ -34,31 +38,40 @@ def execute_salt_command(salt_command: str, shell: bool = False, stdout: Optiona
 
 
 def get_db_cluster(oldwiki_db: str) -> str:
-    command = generate_salt_command('db171*', f'cmd.run "mysql -e \'SELECT wiki_dbcluster FROM mhglobal.cw_wikis WHERE wiki_dbname = "{oldwiki_db}" \' "')
-    result = execute_salt_command(salt_command=command, shell=True, stdout=subprocess.PIPE, text=True)
+    db_query = f'SELECT wiki_dbcluster FROM mhglobal.cw_wikis WHERE wiki_dbname = \\"{oldwiki_db}\\"'
+    command = generate_salt_command("db171", f"sudo -i mysql --skip-column-names -e '{db_query}'")
+    result = execute_salt_command(salt_command=command, stdout=subprocess.PIPE, text=True)
     if result:
-        cluster_name = result.stdout.strip()
-        return db_clusters[cluster_name]  # type: ignore[literal-required]
+         cluster_name = result.stdout.strip()
+         #print(cluster_name)
+         cluster_data = cluster_name.split('\n')
+         cluster_data_b = cluster_data[1].split(' ')
+         print(cluster_data_b)
+         #print("Extracted cluster_name:", cluster_name)  # Print cluster_name for debugging
+         cluster_name = cluster_data_b[4]
+
+         return db_clusters[cluster_name]  # type: ignore[literal-required]
     raise KeyboardInterrupt('Impossible to skip. Aborted.')
 
 
 def rename_wiki(oldwiki_db: str, newwiki_db: str) -> None:
     # Step 1: Get the db cluster for the old wiki dbname
-    oldwiki_cluster = get_db_cluster(oldwiki_db)
 
     try:
         oldwiki_cluster = get_db_cluster(oldwiki_db)
-    except KeyError:
+    except KeyError, IndexError:
         print(f'Error: Unable to determine the db cluster for {oldwiki_db}')
         sys.exit(1)
 
     # Step 2: Execute SQL commands for rename
-    execute_salt_command(salt_command=generate_salt_command(oldwiki_cluster, f'mysqldump {oldwiki_db} > oldwikidb.sql'))
-    execute_salt_command(salt_command=generate_salt_command(oldwiki_cluster, f"mysql -e 'CREATE DATABASE {newwiki_db}'"))
-    execute_salt_command(salt_command=generate_salt_command(oldwiki_cluster, f"mysql -e 'USE {newwiki_db}; SOURCE /home/$user/oldwikidb.sql'"))
+    execute_salt_command(salt_command=generate_salt_command(oldwiki_cluster, f'sudo -i mysqldump {oldwiki_db} > /home/{os.getlogin()}/{oldwiki_db}.sql'))
+    execute_salt_command(salt_command=generate_salt_command(oldwiki_cluster, f"sudo -i mysql -e 'CREATE DATABASE {newwiki_db}'"))
+    execute_salt_command(salt_command=generate_salt_command(oldwiki_cluster, f"sudo -i mysql -e 'USE {newwiki_db}; SOURCE /home/{os.getlogin()}/{oldwiki_db}.sql;'"))
 
     # Step 3: Execute MediaWiki rename script
-    execute_salt_command(salt_command=generate_salt_command('mwtask181', f'sudo -u www-data php /srv/mediawiki/1.41/extensions/CreateWiki/maintenance/renameWiki.php --wiki=loginwiki --rename {oldwiki_db} {newwiki_db} $user'))
+    execute_salt_command(salt_command=generate_salt_command('mwtask181', f'mwscript extensions/CreateWiki/renameWiki.php loginwiki --no-log --rename {oldwiki_db} {newwiki_db} {os.getlogin()}'))
```

Comment:
I think mwtask181 should be some sort of constant so it is easy to change if task server is changed.

Reply:
Yeah that makes sense, how exactly would I do that here?

Reply:
Similar to https://github.com/miraheze/puppet/blob/ac99c896596fa43eb01bb5000db17fa2f8506825/modules/mediawiki/files/bin/mwdeploy.py#L22